shell/platform/darwin/macos/framework/Source/FlutterEngine.mm

```diff
@@ -448,12 +465,12 @@ - (instancetype)initWithName:(NSString*)labelPrefix
   [self setUpAccessibilityChannel];
   [self setUpNotificationCenterListeners];
   id<NSApplicationDelegate> appDelegate = [[NSApplication sharedApplication] delegate];
-  const SEL selector = @selector(addApplicationLifecycleDelegate:);
-  if ([appDelegate respondsToSelector:selector]) {
+  if ([appDelegate conformsToProtocol:@protocol(FlutterAppLifecycleProvider)]) {
```

Comment:
This is a cleaner way to express the check, and also is what the iOS engine does.

Reply:
Yeah less raw selector wrangling is always a plus.